### PR TITLE
CXP-1381: Map asset collection attributes > text (string[])

### PR DESCRIPTION
**Pull request by masacc**
**Description (for Contributor and Core Developer)**

EE: https://github.com/akeneo/pim-enterprise-dev/pull/16702

image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidScopeValidator.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        if (!$constraint instanceof AssetAttributeSourceContainsValidScope) {
+            throw new UnexpectedTypeException($constraint, AssetAttributeSourceContainsValidScope::class);
+        }
+
+        if (!\is_array($value)) {
+            throw new UnexpectedValueException($value, 'array');
+        }
+
+        /** @var AssetAttributeSource $value */
+
```
Comment (guillaumefer-akeneo) [inline components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidScopeValidator.php:42]:
```suggestion

```
remove blank line

**Review by guillaumefer-akeneo — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeExistsValidator.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if (!$constraint instanceof AssetAttributeExists) {
+            throw new UnexpectedTypeException($constraint, AssetAttributeExists::class);
+        }
+
+        if (!\is_array($value)) {
+            throw new UnexpectedValueException($value, 'array');
+        }
+
+        /** @var AssetAttributeSource $value */
+
```
[the remark this answers is not in the document]
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeExistsValidator.php:40]: no, this is on purpose to make phpstan/psalm happy :-)

File: components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidLocaleValidator.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        if (!$constraint instanceof AssetAttributeSourceContainsValidLocale) {
+            throw new UnexpectedTypeException($constraint, AssetAttributeSourceContainsValidLocale::class);
+        }
+
+        if (!\is_array($value)) {
+            throw new UnexpectedValueException($value, 'array');
+        }
+
+        /** @var AssetAttributeSource $value */
+
```
[the remark this answers is not in the document]
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidLocaleValidator.php:44]: same here

File: components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidScopeValidator.php (hunk 1 above)
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Validation/ProductMapping/AssetAttributeSourceContainsValidScopeValidator.php:42]: same here

File: components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributesByTargetTypeAndTargetFormatAction.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                \sprintf(
+                    'The combination of type "%s" and format "%s" does not match any type in the PIM.',
+                    $targetType,
+                    $targetFormat ?? '',
+                ),
+                $exception,
+            );
+        }
+
+        // todo : get PIM current locale
```
Comment (helenegzlz) [inline components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributesByTargetTypeAndTargetFormatAction.php:65]: Another todo to handle.

**Review by eoliveauakeneo — APPROVED** [review]

**Review by helenegzlz — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributeAction.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    ) {
+    }
+
+    public function __invoke(Request $request, string $identifier): Response
+    {
+        if (!$request->isXmlHttpRequest()) {
+            return new RedirectResponse('/');
+        }
+
+        // todo : get PIM current locale
```
[the remark this answers is not in the document]
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributeAction.php:32]: As seen together, it needs some thinking before implementing it. It willy be done in another PR.

File: components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributesByTargetTypeAndTargetFormatAction.php (hunk 4 above)
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Controller/Internal/AssetManager/GetAssetAttributesByTargetTypeAndTargetFormatAction.php:65]: As seen together, it needs some thinking before implementing it. It willy be done in another PR.

**Pull request merged**